packages/core/src/components/button/buttons.tsx

```diff
@@ -80,13 +88,46 @@ export class AnchorButton extends React.Component<React.HTMLProps<HTMLAnchorElem
             </a>
         );
     }
+
+    // Provide consistent keyboard interactions across both <Button /> and <AnchorButton /> (#430)
+    private onKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
+        const { href, onClick, target } = this.props;
+        if (e.which === Keys.SPACE) {
+            e.preventDefault();
+
+            if (href) {
+                if (target === undefined || target === "_self") {
+                    window.open(href);
+                } else if (target === "_blank") {
+                    window.open(href, "_newtab");
+                }
+            } else {
+                this.setState({isActive: true});
+            }
+        }
+
+        if (e.which === Keys.ENTER && this.props.onClick) {
+            onClick(e as any);
```

Comment:
use `safeInvoke` util function instead of `&&` clause. (also avoid boolean coercion.)